core/feconf.py

```diff
@@ -24,11 +24,28 @@
 
 from core.constants import constants
 
-from typing import Dict, List, Union
+from typing import Dict, List
 from typing_extensions import TypedDict
 
-CommandType = (
-    Dict[str, Union[str, List[str], Dict[str, Union[str, List[str]]]]])
+
+# This TypedDict is used only to create the TypedDict below, because we want
+# to mark some attributes as optional.
+class _ValidCmdDictNotTotal(TypedDict, total=False):
+    """Dictionary representing a non-required part
+    of valid commands specs."""
+
+    allowed_values: Dict[str, List[str]]
+    deprecated_values: Dict[str, List[str]]
+
+
+class ValidCmdDict(_ValidCmdDictNotTotal, total=False):
+    """Dictionary representing valid commands specs."""
+
+    name: str
+    required_attribute_names: List[str]
+    optional_attribute_names: List[str]
+    user_id_attribute_names: List[str]
+
```

Comment:
@vojtechjelinek Are we expecting to define custom `type` in feconf.py? if not, do you have any suggestion for a better place to define these global custom types?

Reply:
@div-yam Is there any issue with defining these `type` in change_domain.py file?

Reply:
@DubeySandeep I think it should be fine. We cannot define it in _change_domain.py_ as we also need to use it in _feconf.py_. I think we can have a constants file just for types but I think only if there is more types that need to be placed into _feconf.py_